### DISCUSSION TITLE
Local Sensor data to interface direct

### DISCUSSION THREAD

**Discussion by elik745i** [HASwitchPlate/openHASP#565] [Q&A] [answered]
Is there an option to pass on data from local sensors directly to interface without going around through posting MQTT message and back to interface of the device?
For instance, I got voltage divider hooked to GPIO 35, data within the range of 4095 I got that, I have custom code which converts data read to mapped level I need and posting it to MQTT topıc. I got also bar element setup in interface. Currently I'm posting data to Home Assıstant, then back to hasp device with conversion from 0-100% to 0-255 to be reflected in the bar.
Problem is if there are no connection to MQTT interface will not reflect battery level. So I need to figure out the way to pass on data directly to interface elements. I got that that I can do that over settings configuring GPIO and assigning group to it, but them mapping going to be incorrect.
So the way forward would be somehow passing data streight from my custom code, but how?

**Comment by fvanroie** [comment, marked as answer]
sure, have a look at the custom code examples:
https://github.com/HASwitchPlate/openHASP/blob/6cf4262f2356eaa686abe045ab01a5467a6963b1/src/custom/my_custom_fan_template.cpp#L29-L33

once you have the object pointer you can use the whole [LVGL api v7](https://docs.lvgl.io/7.11/)

**Reply by elik745i** [reply]
Great, thanks that worked!
